src/wast-parser.y

```diff
@@ -650,9 +652,10 @@ expr1 :
       assert(if_->type == ExprType::If);
       if_->if_.true_->label = $2;
     }
-  | try_check LPAR BLOCK labeling_opt block RPAR catch_list {
-      $5->label = $4;
-      Expr* try_ = Expr::CreateTry($5, $7.first);
+  | try_check labeling_opt LPAR BLOCK labeling_opt block RPAR catch_list {
```

Comment:
Yes. Its the "(" shift/reduce conflict problem with the catches. Hence, I made an explicit explicit block for the code in the try block, but that introduced issues concerning the label.

In the parenthesis form, I also would like some type of node clustering the "try block body", similar to the "then" and the "else" in an "if". Hence, I chose a "block" node without a label. Only the try node gets a label.